.github/workflows/intel.yml

```diff
@@ -0,0 +1,114 @@
+name: Intel Linux Build
+on: [push, pull_request, workflow_dispatch]
+
+# Use custom shell with -l so .bash_profile is sourced which loads intel/oneapi/setvars.sh
+# without having to do it in manually every step
+defaults:
+  run:
+    shell: bash -leo pipefail {0}
+
+# Set I_MPI_CC/F90 so Intel MPI wrapper uses icc/ifort instead of gcc/gfortran
+env:
+  cache_key: intel1  # The number (#) following the cache_key "intel" is to flush Action cache.
+  CC: icc
+  FC: ifort
+  CXX: icpc
+  I_MPI_CC: icc
+  I_MPI_F90: ifort
+
+# A note on flushing Action cache and relevance to "cache_key" above.
+# There is no way to flush the Action cache, and hence a number (#) is appended
+# to the "cache_key" (intel).
+# If the dependencies change, increment this number and a new cache will be
+# generated by the dependency build step "setup"
+# There is a Github issue to force clear the cache.
+# See discussion on:
+# https://stackoverflow.com/questions/63521430/clear-cache-in-github-actions
```

Comment:
Eventually using a hash of  the `spack.lock` file would be useful to detect changes in the dependencies

Reply:
Let me know when you have that.